Dynatrace/dynatrace-operator
# Updates the image csi-gc to clean up unused shared images

# Description
When the `codeModulesImage` is used then shared images are stored/unpacked under `/data/codemodules/<image-digest>`

To clean these folders up we check in the metadata database for the used image digests. (stored in the `dynakube table` + the versions in the `volumes` table)
We compare that to the folders we see, and if the folder's name is an image-digest that is not present in the database than we delete it.

~***Important sidenote***: I think the current implementation needs some changes, because if we delete a dynakube that has the digest that is used by a few volumes, we would delete the lowerdir from under them. In this case I don't know what will happen, I still have to test that.~ (fixed in: https://github.com/Dynatrace/dynatrace-operator/pull/847/commits/17c87ad2d3041aff9f8ad31f8b812056095fd54a)

## How can this be tested?
Have a set of dynakubes with `codeModulesImage` that point to images that have different digests.
Deploy them, wait for the downloads to finish, then delete one of the dynakubes.
Than check if `/data/codemodules/<image-digest>` was removed successfully.
**Important sidenote**:  The current GC logic is that it only runs if at least 1 dynakube is in the cluster. (which is not optimal IMO, but fixing that is not trivial and out of scope for this change)

## Checklist
